core-contracts/FeeHandler/src/main/java/network/balanced/score/core/feehandler/FeeHandlerImpl.java

```diff
@@ -198,11 +256,23 @@ public void tokenFallback(Address _from, BigInteger _value, byte[] _data) {
         lastTxhash.set(Context.getTransactionHash());
         lastFeeProcessingBlock.set(sender, BigInteger.valueOf(Context.getBlockHeight()));
 
-        for (int i = 0; i < acceptedDividendsTokens.size(); i++) {
+        int acceptedDividendsTokensCount = acceptedDividendsTokens.size();
+        for (int i = 0; i < acceptedDividendsTokensCount; i++) {
             if (acceptedDividendsTokens.get(i).equals(sender)) {
                 transferToken(sender, getContractAddress("dividends"), getTokenBalance(sender), new byte[0]);
             }
         }
+        if (_from.equals(loans.get())) {
+            loanFeesAccrued.set(loanFeesAccrued.getOrDefault(BigInteger.ZERO).add(_value));
+        }
+
+        BigInteger accruedFees = swapFeesAccruedDB.get(sender);
+        if (_from.equals(dex.get()) && accruedFees != null) {
+            swapFeesAccruedDB.set(sender, accruedFees.add(_value));
+        }
+        if (_from.equals(stabilityFund.get())) {
+            stabilityFundFeesAccrued.set(stabilityFundFeesAccrued.getOrDefault(BigInteger.ZERO).add(_value));
+        }
```

Comment:
Since this affect basically all transactions we should optimize gas a little
```suggestion
        if (_from.equals(dex.get())) {
            BigInteger accruedFees = swapFeesAccruedDB.get(sender);
            if (accuredFees == null) {
                return;
            }
            
            swapFeesAccruedDB.set(sender, accruedFees.add(_value));
            return;
        }
        
        if (_from.equals(loans.get())) {
            loanFeesAccrued.set(loanFeesAccrued.getOrDefault(BigInteger.ZERO).add(_value));
            return;
        }
    
        if (_from.equals(stabilityFund.get())) {
            stabilityFundFeesAccrued.set(stabilityFundFeesAccrued.getOrDefault(BigInteger.ZERO).add(_value));
            return;
        }
```

Reply:
Better to use if-else ladder or switch.